src/Views/AppInfoView.vala

```diff
@@ -281,32 +281,48 @@ namespace AppCenter.Views {
             new Thread<void*> ("content-loading", () => {
                 app_version.label = package.get_version ();
 
-                string url = null;
-                uint max_size = 0U;
-
                 if (screenshots.length == 0) {
                     return null;
                 }
 
+                Gee.Deque<Gtk.Widget> screenshot_widgets = new Gee.LinkedList<Gtk.Widget> ();
+
                 screenshots.foreach ((screenshot) => {
-                    screenshot.get_images ().foreach ((image) => {
-                        if (max_size < image.get_width ()) {
-                            url = image.get_url ();
-                            max_size = image.get_width ();
+                    Gtk.Widget screenshot_widget = null;
+                    var index = 0;
+                    var images = screenshot.get_images ();
+
+                    while (screenshot_widget == null && index < images.length) {
```

Comment:
foreach's are easier to understand :) You can add a `break` if the `widget != null`, and you also avoid having to declare the index